xwiki-platform-core/xwiki-platform-security/xwiki-platform-security-authentication/xwiki-platform-security-authentication-api/src/main/java/org/xwiki/security/authentication/ResetPasswordManager.java

```diff
@@ -77,4 +78,17 @@ ResetPasswordRequestResponse checkVerificationCode(UserReference userReference,
      */
     void resetPassword(UserReference userReference, String newPassword)
         throws ResetPasswordException;
+
+    /**
+     * Ensure that the password matches the requirements provided by {@link RegistrationConfiguration}.
+     *
+     * @param newPassword the password to check
+     * @return {@code true} if the rules exposed in the {@link RegistrationConfiguration} are all respected.
+     * @since 15.10RC1
+     */
+    @Unstable
+    default boolean isPasswordCompliantWithRegistrationRules(String newPassword)
+    {
+        return true;
```

Comment:
Small improvement suggestion. Could be interesting to mention in the javadoc that the default is very permissive.